firmware/src/networking/wifi/esp.rs

```diff
@@ -0,0 +1,74 @@
+extern crate alloc;
+use alloc::format;
+use core::str;
+
+use defmt::{debug, error, info, trace};
+use embassy_executor::task;
+use embassy_futures::yield_now;
+use embedded_svc::ipv4::Interface;
+use esp_wifi::{
+	create_network_stack_storage, current_millis, network_stack_storage,
+	wifi::utils::create_network_interface, wifi_interface::{Network, IoError as WifiError},
+};
+use smoltcp::socket::UdpPacketMetadata;
+
+#[task]
+pub async fn network_task() {
+	let mut storage = create_network_stack_storage!(3, 8, 1, 1);
+	let ethernet = create_network_interface(network_stack_storage!(storage));
+	let mut wifi = esp_wifi::wifi_interface::Wifi::new(ethernet);
```

Comment:
It looks to me like from here on out, all of this code is no longer platform specific. I have created #82 to track this.

Reply:
DHCP is still hardware specific sadly